src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2804,6 +2819,178 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
+
+    auto& root = app->getLedgerTxnRoot();
+
+    auto entries = LedgerTestUtils::generateValidUniqueLedgerEntriesWithTypes(
+        {CONTRACT_CODE, CONTRACT_DATA}, cfg.ENTRY_CACHE_SIZE + 1);
+    std::set<LedgerEntry> entrySet;
+    std::map<LedgerKey, LedgerEntry> entryMap;
+    LedgerTxn ltx(root);
+    auto lkMeter = std::make_shared<LedgerKeyMeter>();
+    auto lkMeter2 = std::make_shared<LedgerKeyMeter>();
+    LedgerKeySet expectedSuccessKeys;
+    LedgerKeySet expectedFailedKeys;
+    for (auto e : entries)
+    {
+        ltx.createWithoutLoading(e);
+        keysToPrefetch.emplace(LedgerEntryKey(e));
+        UnorderedSet<LedgerKey> ks{};
+        ks.emplace(LedgerEntryKey(e));
+        stellar::uniform_int_distribution<uint32_t> dist(1, 10);
+        int32_t offset = 5 - dist(stellar::gRandomEngine);
+        lkMeter->addTxn(xdr::xdr_size(e) + offset, ks);
+        if (offset < 0)
+        {
+            REQUIRE(!lkMeter->canLoad(LedgerEntryKey(e), xdr::xdr_size(e)));
+            expectedFailedKeys.emplace(LedgerEntryKey(e));
+        }
+        else
+        {
+            REQUIRE(lkMeter->canLoad(LedgerEntryKey(e), xdr::xdr_size(e)));
+            expectedSuccessKeys.emplace(LedgerEntryKey(e));
+        }
+
+        // Insert entry into set with correct lastModifiedLedgerSeq value so
+        // we can check prefetch results later
+        e.lastModifiedLedgerSeq = 1;
+        entrySet.emplace(e);
+        entryMap.emplace(LedgerEntryKey(e), e);
+    }
+    if (cfg.EXPERIMENTAL_BUCKETLIST_DB)
+    {
+        std::vector<LedgerEntry> ledgerVect{entrySet.begin(), entrySet.end()};
+        app->getBucketManager().addBatch(*app, 2, 20, {}, ledgerVect, {});
+    }
+    ltx.commit();
+
+    SECTION("prefetch normally")
+    {
+        LedgerTxn ltx2(root);
+        UnorderedSet<LedgerKey> smallSet;
+        LedgerKeySet expectedSuccessKeys2;
+        LedgerKeySet expectedFailedKeys2;
+
+        for (auto const& k : keysToPrefetch)
+        {
+            auto entry = entryMap.at(k);
+            UnorderedSet<LedgerKey> ks{k};
+            stellar::uniform_int_distribution<uint32_t> dist(1, 10);
+            int32_t offset = 5 - dist(stellar::gRandomEngine);
+            // store the initial quota for the transaction.
+            auto quota = xdr::xdr_size(entry) + offset;
+            lkMeter2->addTxn(quota, ks);
+            if (offset < 0)
+            {
+                expectedFailedKeys2.emplace(k);
+                REQUIRE(!lkMeter2->canLoad(k, xdr::xdr_size(entry)));
+            }
+            else
+            {
+                expectedSuccessKeys2.emplace(k);
+                REQUIRE(lkMeter2->canLoad(k, xdr::xdr_size(entry)));
+            }
+            smallSet.emplace(k);
+            if (smallSet.size() > (cfg.ENTRY_CACHE_SIZE / 3))
+            {
+                break;
+            }
+        }
+        auto loadSize = root.prefetchSoroban(smallSet, lkMeter2);
+        for (auto const& k : expectedFailedKeys2)
```

Comment:
This check is redundant with the `if (expectedFailedKeys2.find(k) != expectedFailedKeys2.end())` check below.